man/systemd-repart.xml

```diff
@@ -374,6 +364,16 @@
         <xi:include href="version-info.xml" xpointer="v252"/></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--signing-engine=</option></term>
+
+        <listitem><para>Takes the name of a provider or engine, that will be passed to OpenSSL's
+        "engine" or "provider" logic. Configures the signing mechanism to use when creating verity signature
+        partitions with the <varname>Verity=signature</varname> setting in partition files.</para>
```

Comment:
i don't think we should add new setting called "engine" if that's the concept that is on its way out. Hence, please name this "provider" please, even if it will possibly invoke an engine in some cases.

Please update the explanation of the --private-key= option and clarify that it typically takes a file path, but if the --signing-engine= option is used might also take a URI.

Reply:
"on its way out" seems wishful thinking to me. There are no production quality providers, and nobody seems to use them. Every other existing tool and documentation and script calls them engines, including our own like ukify. If they actually become a usable thing at some point in the future, we can rename them.

Reply:
Naming this `--signing-provider=` doesn't make sense though? Since when using providers the actual provider used is part of the URI. Only when using engines the "engine" is supplied separately.

So maybe a better way would be to have `--signing-mode=file|engine|provider`, `--signing-engine=` which is only used when `--signing-mode=engine` and `--private-key=` which takes a file, an engine URI or a provider URI depending on `--signing-mode=`.

Reply:
It would be redundant though, if I pass an --engine then it's because I want that engine to be used. Also no other tool uses such as scheme AFAIK